### PR TITLE
Feature/react 81 react router setup

### PR DESCRIPTION
**Pull request by mayuriardad**

### CODE REVIEW

File: react-frontend/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "react-bootstrap": "^1.0.0",
     "react-dom": "^16.13.1",
     "react-redux": "^7.2.0",
+    "react-router-dom": "^5.1.2",
```
Comment (sahilbhatia) [inline react-frontend/package.json:17]: Please remove the empty spaces

File: react-frontend/src/App.js [hunk 2]
```diff
@@ -1,23 +1,16 @@
...
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <p>Welcome to peerly</p>
```
Comment (sahilbhatia) [inline react-frontend/src/App.js:12]: `P` should be capital in `Peerly`

File: react-frontend/src/App.js [hunk 3]
```diff
@@ -1,23 +1,16 @@
 import React from "react";
+import { Link } from "react-router-dom";
+
 import logo from "./logo.svg";
-import "./App.css";
+import "App.css";
```
Comment (sahilbhatia) [inline react-frontend/src/App.js:5]: Please remove this, as we will be using styled components

File: react-frontend/src/App.js (hunk 2 above)
Comment (sahilbhatia) [inline react-frontend/src/App.js:12]: Project name should be in title case

File: react-frontend/src/App.js [hunk 4]
```diff
@@ -1,23 +1,16 @@
...
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <p>Welcome to peerly</p>
+        <Link to="/dashboard">dashboard</Link>
```
Comment (sahilbhatia) [inline react-frontend/src/App.js:13]: Move all route paths to a constants file.

File: react-frontend/src/Dashboard.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+import "./App.css";
```
Comment (sahilbhatia) [inline react-frontend/src/Dashboard.js:3]: Please remove this, as we will be using styled components

File: react-frontend/src/HOC/withLayout.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+import Topnav from "Topnav";
```
Comment (sahilbhatia) [inline react-frontend/src/HOC/withLayout.js:3]: Why is `Topnav` component placed directly in `src` directory (and not inside `components`)?

File: react-frontend/src/HOC/withLayout.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import Sidebar from "Sidebar";
```
Comment (sahilbhatia) [inline react-frontend/src/HOC/withLayout.js:4]: Why is `Sidebar` component placed directly in `src` directory (and not inside `components`)?

File: react-frontend/src/Sidebar.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+import "./App.css";
```
Comment (sahilbhatia) [inline react-frontend/src/Sidebar.js:3]: Please remove this, as we are going to use styled components

File: react-frontend/src/reducers/appReducer.js [hunk 9]
```diff
@@ -1,4 +1,6 @@
-const defaultState = {};
+const defaultState = {
+  token: "",
```
Comment (sahilbhatia) [inline react-frontend/src/reducers/appReducer.js:2]: Make default value `null`. Doing so means developers ensure that code handles null values.

File: react-frontend/src/root-routes-config.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import App from "App";
+import Dashboard from "Dashboard";
+import UnathorisedFallback from "sharedComponents/UnathorisedFallback";
```
Comment (sahilbhatia) [inline react-frontend/src/root-routes-config.js:3]: Don't use the term `fallback` in name.

File: react-frontend/src/root-routes-config.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+export default [
+  {
+    path: "/",
+    is_protected: false,
+    is_navbar: false,
```
Comment (sahilbhatia) [inline react-frontend/src/root-routes-config.js:9]: Naming convention is wrong. Please rename to `includeNavbar`

File: react-frontend/src/root-routes-config.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    is_sidebar: false,
```
Comment (sahilbhatia) [inline react-frontend/src/root-routes-config.js:10]: Naming convention is wrong. Please rename to `includeSidebar`

File: react-frontend/src/root-routes-config.js [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import App from "App";
+import Dashboard from "Dashboard";
+import UnathorisedFallback from "sharedComponents/UnathorisedFallback";
+
+export default [
+  {
+    path: "/",
```
Comment (sahilbhatia) [inline react-frontend/src/root-routes-config.js:7]: For re-usability, move all path names to a constants hash

File: react-frontend/src/root-routes.js [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import { Switch } from "react-router-dom";
+
+import GenerateRoutes from "sharedComponents/GenerateRoutes";
+import rootRoutesConfig from "root-routes-config";
+
+function Routes() {
+  return <Switch>{<GenerateRoutes configs={rootRoutesConfig} />}</Switch>;
```
Comment (sahilbhatia) [inline react-frontend/src/root-routes.js:8]: Should be singular - `config`. As it's a single config object

File: react-frontend/src/sharedComponents/GenerateRoutes.js [hunk 15]
```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { Route } from "react-router-dom";
+import PropTypes from "prop-types";
+
+import PrivateRoute from "sharedComponents/PrivateRoute";
+
+function GenerateRoutes({ configs }) {
+  return (
+    <>
```
Comment (sahilbhatia) [inline react-frontend/src/sharedComponents/GenerateRoutes.js:9]: I believe fragment is not required here. Please confirm once

File: react-frontend/src/sharedComponents/GenerateRoutes.js [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+      {configs.map((route) => {
```
Comment (sahilbhatia) [inline react-frontend/src/sharedComponents/GenerateRoutes.js:10]: No inline functions within render/return.
Within the same file,  move it to a separate function (outside this component definition)

File: react-frontend/src/root-routes-config.js [hunk 17]
```diff
@@ -0,0 +1,30 @@
+import App from "App";
+import Dashboard from "Dashboard";
+import UnathorisedFallback from "sharedComponents/UnathorisedFallback";
+
+export default [
```
Comment (sahilbhatia) [inline react-frontend/src/root-routes-config.js:5]: Include `strict` and `key` arguments

File: react-frontend/src/sharedComponents/GenerateRoutes.js [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+            key={route.path}
+          />
+        );
+      })}
+    </>
+  );
+}
+
+GenerateRoutes.propTypes = {
+  configs: PropTypes.arrayOf(
```
Comment (sahilbhatia) [inline react-frontend/src/sharedComponents/GenerateRoutes.js:27]: 1. Missing `required`
2. Missing default values

File: react-frontend/src/sharedComponents/SessionFallback.js [hunk 19]
```diff
@@ -0,0 +1,18 @@
+import React, { useEffect } from "react";
+import { useHistory } from "react-router-dom";
+
+const redirectTimeout = 3000;
```
Comment (sahilbhatia) [inline react-frontend/src/sharedComponents/SessionFallback.js:4]: Move this to global constants

File: react-frontend/src/HOC/withLayout.js [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+
+const withLayout = (Component) => (props) => {
+  return (
+    <>
+      <Topnav />
```
Comment (sahilbhatia) [inline react-frontend/src/HOC/withLayout.js:9]: Shouldn't we be using the flag `includeTopnav` here?

File: react-frontend/src/HOC/withLayout.js [hunk 21]
```diff
@@ -0,0 +1,16 @@
...
+      <Component {...props} />
+      <Sidebar />
```
Comment (sahilbhatia) [inline react-frontend/src/HOC/withLayout.js:11]: Shouldn't we be using the flag `Sidebar` here?

File: react-frontend/package.json (hunk 1 above)
Comment (mayuriardad) [reply react-frontend/package.json:17]: sorry I don't get this which empty spaces?

**Pull request merged**